### ISSUE TITLE
Clarify linux (ubuntu 16.04) installation

### ISSUE THREAD

**Issue by reece** [jaraco/keyring#223]
When following the README on Ubuntu 16.04 (GNOME desktop, not headless), I get `RuntimeError: No recommended backend was available. Install the keyrings.alt package if you want to use the non-recommended backends. See README.rst for details.`.

I understand what this is saying, but not what to do about it. I infer from the README that this should work by default. (For this reason, I didn't install keyrings.alt, but did notice that it has a Gnome backend.)

```
These recommended keyring backends are supported by the Python keyring lib:

    Mac OS X Keychain
    Linux Secret Service (requires secretstorage)
    KWallet (requires dbus)
    Windows Credential Vault

Other keyring implementations are provided in the keyrings.alt package.
```

If you can send me down the right path, I'll be happy to document what I do and submit a documentation PR.

To reproduce:

```
snafu$ cd /tmp
snafu$ pyvenv py3
snafu$ source py3/bin/activate
(py3) snafu$ pip install -U pip keyring secretstorage
Collecting pip
  Using cached pip-8.1.2-py2.py3-none-any.whl
Collecting keyring
  Using cached keyring-9.0-py2.py3-none-any.whl
Collecting secretstorage
Collecting pycrypto (from secretstorage)
Installing collected packages: pip, keyring, pycrypto, secretstorage
  Found existing installation: pip 8.1.1
    Uninstalling pip-8.1.1:
      Successfully uninstalled pip-8.1.1
Successfully installed keyring-9.0 pip-8.1.2 pycrypto-2.6.1 secretstorage-2.1.4
(py3) snafu$ python
Python 3.5.1+ (default, Mar 30 2016, 22:46:26) 
[GCC 5.3.1 20160330] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import keyring
>>> keyring.set_password("system", "username", "password")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/tmp/py3/lib/python3.5/site-packages/keyring/core.py", line 48, in set_password
    _keyring_backend.set_password(service_name, username, password)
  File "/tmp/py3/lib/python3.5/site-packages/keyring/backends/fail.py", line 18, in get_password
    raise RuntimeError("No recommended backend was available. Install the "
RuntimeError: No recommended backend was available. Install the keyrings.alt package if you want to use the non-recommended backends. See README.rst for details.
```

**Comment by jaraco** [comment]
Hey reece. Sorry for the trouble. What you've done there looks exactly right. I was able to replicate your error on Ubuntu Trusty on Python 3.4. After installing the packages similar to how you did (I had to `apt-get install -y python3-dev` in order to build pycrypto), I found that secretstorage appears broken:

```
vagrant@trusty:~$ env/bin/python
Python 3.4.3 (default, Oct 14 2015, 20:28:29) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import secretstorage
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/vagrant/env/lib/python3.4/site-packages/secretstorage/__init__.py", line 13, in <module>
    import dbus
ImportError: No module named 'dbus'
```

So it seems secretstorage has an undeclared dependency on dbus. Or rather, it's [declared in the package long description](https://pypi.python.org/pypi/SecretStorage), but it's not clear how one gets it. Best I can tell, there are no installation instructions, [even in the docs](https://dbus.freedesktop.org/doc/dbus-python/) and [the project readme](https://cgit.freedesktop.org/dbus/dbus-python/tree/README).

I'm pretty sure if you can get dbus installed, things will start to work. I wish it weren't so much trouble, but such is the nature of Linux. If appreciate the offer to help with docs, either as a troubleshooting section or other tips on installing on Linux. I welcome the contribution.

**Comment by havocp** [comment]
Wow. I had this problem just now and found a half-hour-old discussion. :-)

It looks like `pip install dbus-python` fixes it for me, on Fedora.

**Comment by reece** [comment]
Thanks, @jaraco and @havocp. Indeed, installing dbus solved the issue. PR coming as promised...

Thanks for keyring. I'd like to see this catch on!

**Comment by mitya57** [comment]
python-dbus used to be not installable via PyPI, so secretstorage could not depend on it. Now that it is installable via PyPI, I have added the dependency in secretstorage git, that change will be in next release.

In any case installing the distro packages for secretstorage and/or python-dbus is probably easier than compiling the latter from sources.

While we are talking about dependencies: maybe it makes sense to make Keyring depend on secretstorage on Linux?

**Comment by jaraco** [comment]
Sounds like a good idea to me.

**Comment by mitya57** [comment]
> python-dbus used to be not installable via PyPI, so secretstorage could not depend on it. Now that it is installable via PyPI, I have added the dependency in secretstorage git, that change will be in next release.

I have now released SecretStorage 2.2.0, which pulls dbus-python when installing via pip/setuptools.
